cc/utils/cose/cose.cc

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2024 The Project Oak Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "cc/utils/cose/cose.h"
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "absl/status/statusor.h"
+#include "absl/strings/string_view.h"
+#include "libcppbor/include/cppbor/cppbor.h"
+#include "libcppbor/include/cppbor/cppbor_parse.h"
+
+namespace oak::utils::cose {
+
+absl::StatusOr<CoseSign1> CoseSign1::Deserialize(const std::vector<uint8_t>& data) {
```

Comment:
Prefer `absl::Span`, unless the cppbor code insists this be a reference to a vector.

Reply:
It's fine -- Span is a contiguous region of memory -- but we'll most likely deal with vectors anyway, so it's not a big deal. (Otherwise, you could call `parse(span.data(), span.size())'`)

Reply:
Parsing protobufs will produce an `std::string`, is there a conversion between those? Or it's better to use `absl::string_view` instead of span? (I was hitting zero-byte related problems when trying to use string view)

Reply:
Changed input arguments to `absl::string_view`